keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,687 @@
+
+### Goals
+
+- Provide a privileged mode for cluster administrators to access devices already
+  in use.
+
+- Ensure that only authorized users can use the privileged mode.
+
+- Maintain security and prevent unauthorized access to devices.
+
+### Non-Goals
+
+- Redesign DRA.
+
+- Extend privileged mode access to non-admin users.
+
+- Define how administrators use the data accessed through privileged mode.
+
+- Define which specific additional permissions are enabled by DRA drivers when
+  this privileged mode is used.
+
+## Proposal
+
+### DRAAdminAccess Overview
+
+This proposal enhances the `DRAAdminAccess` feature to enable cluster
+administrators to mark requests in `ResourceClaim` and `ResourceClaimTemplate`
+objects as privileged. This feature includes:
+
+1. Admin Access Flag:
+
+   A field in `ResourceClaim` and `ResourceClaimTemplate` specifications to
+   indicate the request is in admin mode.
+
+   Example:
+
+   ```yaml
+   apiVersion: resource.k8s.io/v1beta1
+   kind: ResourceClaim
+   metadata:
+     name: admin-resource-claim
+     namespace: admin-namespace
+   spec:
+     resourceClassName: admin-resource-class
+     adminAccess: true
+   ```
+
+1. Namespace Label for DRA Admin Mode:
+
+   Namespaces must be labeled to allow the creation of admin-access claims.
+
+   Example label:
+
+   ```yaml
+   metadata:
+     labels:
+       kubernetes.io/dra-admin-access: "true"
+   ```
+
+1. Authorization Check:
+
+   In the REST storage layer, validate requests to create `ResourceClaim` or
+   `ResourceClaimTemplate` objects with `adminAccess: true`. Only authorize if
+   namespace has the `kubernetes.io/dra-admin-access` label.
+
+1. Grants privileged access to the requested device:
+
+   For requests with `adminAccess: true`, the scheduler bypasses standard
+   allocation checks and allows administrators to access devices already in use.
+   This ensures privileged tasks like monitoring or diagnostics can be performed
+   without disrupting existing allocations.
+
+1. No impact on availability of claims:
+
+   The scheduler ignores claims with `adminAccess: true`, normal usage is not
+   impacted as claims in other namespaces can still be allocated using the same
+   devices that are also accessed by workloads in the admin namespace.
+
+### Workflow
+
+1. A cluster administrator labels an admin namespace with
+   `kubernetes.io/dra-admin-access`.
+
+1. Only authorized users can create `ResourceClaim` or `ResourceClaimTemplate`
+   objects with `adminAccess: true` in this admin namespace.
+
+1. Only users with access to the admin namespace can reference these
+   `ResourceClaims` or `ResourceClaimTemplates` in their pod or deployment
+   specs.
+
+1. The DRA controller processes the request and grants privileged access to the
+   requested device.
+
+## Design Details
+
+### API Changes
+
+The `adminAccess` field in `DeviceRequest` was added to `ResourceClaim` and
```

Comment:
A mermaid diagram that shows how this `bool` flows through the entire stack would be helpful.